planner/cascades/transformation_rules.go

```diff
@@ -1448,3 +1452,174 @@ func (r *TransformLimitToTableDual) OnTransform(old *memo.ExprIter) (newExprs []
 	tableDualExpr := memo.NewGroupExpr(tableDual)
 	return []*memo.GroupExpr{tableDualExpr}, true, true, nil
 }
+
+// EliminateOuterJoinBelowAggregation eliminate the outer join which below aggregation
+type EliminateOuterJoinBelowAggregation struct {
+	baseRule
+}
+
+// NewRuleTransformLimitToTableDual creates a new Transformation EliminateOuterJoinBelowAggregation.
+// The pattern of this rule is `Aggregation->Join->X`.
+func NewRuleEliminateOuterJoinBelowAggregation() Transformation {
+	rule := &EliminateOuterJoinBelowAggregation{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandAggregation,
+		memo.EngineAll,
+		memo.NewPattern(memo.OperandJoin, memo.EngineAll),
```

Comment:
Have you ever consider the `Agg->Selection->Join` situations?

Reply:
if selection only use the cols from outer group.
selection will be push down outer join.
`Agg->Selection->Join` will be convert `Agg->Join`

Reply:
in cascades, we can not match pattern more than two level?

Reply:
That's just an example. What about `Agg->projection->join`?

Reply:
`Agg->projection->join` can noly create due to transformation rule applied.
the sql have `join` without any rule applied must be `projection->join`, `Agg->join`, `Selection->join`.
and in cascades, we can find these pattern in group .
isn't it?

Reply:
`Agg->x1->x2->x3...->xN->join`.
if x1,x2....xN all can be eliminated, `Agg->x1->x2->x3...->xN->join` will be converted to `Agg->join` by other Rule.
so i just consider `Agg->join` situations。

such as `Agg->Selection->Join` will be converted to `Agg->join`.
`Agg->Projection-Join>`, 
As for `Agg->projection->join` , I think that we should eliminate `projection`, convert it to `Agg->join` in another rule.